jobapi/routes.go

```diff
@@ -0,0 +1,134 @@
+		protectedEnv[p] = struct{}{}
+	}
+}
+
+func (s *Server) router() chi.Router {
+	r := chi.NewRouter()
+	r.Use(
+		middleware.Recoverer,
+		// middleware.Logger, // REVIEW: Should we log requests to this API? If so, where should we log them to? The job logs?
+		HeadersMdlw(),
+		AuthMdlw(s.token),
+	)
+
+	r.Route("/api/current-job/v0", func(r chi.Router) {
+		r.Get("/env", s.getEnv())
+		r.Patch("/env", s.patchEnv())
+		r.Delete("/env", s.deleteEnv())
+	})
+
+	return r
+}
+
+func (s *Server) getEnv() http.HandlerFunc {
+	return func(w http.ResponseWriter, _ *http.Request) {
+		normalizedEnv := s.environ.Dump()
+		resp := EnvGetResponse{Env: normalizedEnv}
+		w.WriteHeader(http.StatusOK)
+		json.NewEncoder(w).Encode(resp)
+	}
+}
+
+func (s *Server) patchEnv() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var req EnvUpdateRequest
+		err := json.NewDecoder(r.Body).Decode(&req)
+		defer r.Body.Close()
+		if err != nil {
+			err = fmt.Errorf("failed to decode request body: %w", err)
+			returnError(w, err, http.StatusBadRequest)
+			return
+		}
+
+		added := make([]string, len(req.Env))
+		updated := make([]string, len(req.Env))
+		protected := checkProtected(maps.Keys(req.Env))
+
+		if len(protected) > 0 {
+			err = fmt.Errorf("the following environment variables are protected, and cannot be modified: % v", protected)
+			returnError(w, err, http.StatusUnprocessableEntity)
+			return
+		}
+
+		for k, v := range req.Env {
+			if _, ok := s.environ.Get(k); ok {
+				updated = append(updated, k)
+			} else {
+				added = append(added, k)
+			}
+			s.environ.Set(k, v)
+		}
+
+		resp := EnvUpdateResponse{
+			Added:   added,
+			Updated: updated,
+		}
+
+		json.NewEncoder(w).Encode(resp)
+		w.WriteHeader(http.StatusOK)
+	}
+}
+
+func (s *Server) deleteEnv() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var req EnvDeleteRequest
+		err := json.NewDecoder(r.Body).Decode(&req)
+		defer r.Body.Close()
+		if err != nil {
+			err = fmt.Errorf("failed to decode request body: %w", err)
+			returnError(w, err, http.StatusBadRequest)
+			return
+		}
+
+		protected := checkProtected(req.Keys)
+		if len(protected) > 0 {
+			err = fmt.Errorf("the following environment variables are protected, and cannot be modified: % v", protected)
+			returnError(w, err, http.StatusUnprocessableEntity)
+			return
+		}
+
+		deleted := make([]string, 0, len(req.Keys))
+		for _, k := range req.Keys {
+			if _, ok := s.environ.Get(k); ok {
+				deleted = append(deleted, k)
+				delete(*s.environ, k)
+			}
+		}
+
+		json.NewEncoder(w).Encode(EnvDeleteResponse{Deleted: deleted})
+		w.WriteHeader(http.StatusOK)
+	}
+}
+
+func checkProtected(candidates []string) []string {
+	protected := make([]string, 0, len(candidates))
+	for _, c := range candidates {
+		if _, ok := protectedEnv[c]; ok {
+			protected = append(protected, c)
+		}
+	}
+	return protected
+}
+
+func returnError(w http.ResponseWriter, err error, code int) {
```

Comment:
this should potentially be renamed to `writeError` or something, it doesn't return anything and having 
```Go
returnError(w, errors.New("blah"), 500)
return
```
repeats itself and looks a little funny